src/rpdk/core/contract/resource_client.py

```diff
@@ -49,6 +57,24 @@ def prune_properties(document, paths):
     return document
 
 
+def _prune_properties_for_all_sequence_members(document, path):
+    try:
+        # this returns multiple paths
+        _prop, resolved_paths = traverse_path_for_sequence_members(document, path)
+    except LookupError:
+        pass  # not found means nothing to delete
+    else:
+        # paths with indices are gathered in increasing order, but we need to delete in reverse order
+        resolved_paths = resolved_paths[::-1]
+        for resolved_path in resolved_paths:
+            new_doc = document
+            for key in resolved_path[: len(resolved_path) - 1]:
+                new_doc = new_doc[key]
+            key = resolved_path[-1]
+            del new_doc[key]
```

Comment:
The pruning of the attributes happens a little differently, as `traverse_path_for_sequence_members` does not return a parent, and returns multiple resolved_paths. Due to the indices being gathered in sequential order, we remove attributes in reverse order to avoid IndexError.